File: F/2011-final-value-of-variable-after-performing-operations/main.go
package main

import "fmt"

func finalValueAfterOperations(operations []string) int {
	x := 0

	for _, oper := range operations {
		for _, b := range oper {
			if byte(b) == byte('+') {
				x++
				break
			} else if byte(b) == byte('-') {
				x--
				break
			}
		}
	}

	return x
}

func main() {
	fmt.Println()
}

File: F/2011-final-value-of-variable-after-performing-operations/main.rs
struct Solution {}

impl Solution {
  pub fn final_value_after_operations(operations: Vec<String>) -> i32 {
    operations.iter().fold(0, |aggr, oper| {
      aggr
        + if oper.chars().nth(1).unwrap() == '+' {
          1
        } else {
          -1
        }
    })
  }

  pub fn final_value_after_operations2(operations: Vec<String>) -> i32 {
    let mut ret: i32 = 0;
    for oper in operations {
      if oper.as_bytes()[1] == '+' as u8 {
        ret += 1;
      } else {
        ret -= 1;
      }
    }
    ret
  }
}
